modules/drivers/athena/test/metabase/test/data/athena.clj

```diff
@@ -177,11 +216,26 @@
       (log/infof "The following Athena databases have already been created: %s" (pr-str (sort dbs)))
       dbs)))
 
+(def ^:private ^:dynamic *allow-database-creation*
+  "Whether to allow database creation. This is normally disabled to prevent people from accidentally loading duplicate
+  data into Athena or somehow stomping over existing databases and breaking CI. If you want to create a new dataset,
+  change this flag to true and run your code again so the data will be loaded normally. Set it back to false when
+  you're done."
+  false)
+
 (defmethod tx/create-db! :athena
   [driver {:keys [database-name], :as db-def} & options]
   (let [database-name (ddl.i/format-name driver database-name)]
-    (if (contains? (existing-databases) database-name)
+    (cond
+      (contains? (existing-databases) database-name)
       (log/infof "Athena database %s already exists, skipping creation" (pr-str database-name))
+
+      (not *allow-database-creation*)
+      (log/fatalf (str "Athena database creation is disabled: not creating database %s. Tests will likely fail.\n"
+                       "See metabase.test.data.athena/*allow-database-creation* for more info.")
+                  (pr-str database-name))
```

Comment:
Under normal usage we should never get here, since the datasets we're testing against should already exist and come back from `(existing-databases)`. But if you defined a new dataset that doesn't exist yet, you would get here, so I made the log level here `fatalf` to make sure it pops up even in CI to tell you that you have to manually enable dataset creation.

I don't love making it hard to define new datasets in this way -- I'm sure it's only a matter of time before someone adds a new one and then can't figure out why Athena tests aren't passing -- but at least this message should point them in the right direction. And since it seems like the choice is either a) make it harder to define new datasets or b) run the risk of CI randomly exploding again with duplicate Athena data, I think option A is the lesser of two evils here.